packages/openapi-generator/src/file-serializer/__snapshots__/api-file.spec.ts.snap

```diff
@@ -72,33 +78,78 @@ import type { QueryParameterType, RefType, ResponseType } from './schema';
  * This API is part of the 'MyServiceName' service.
  */
 export const TestApi = {
+  _defaultBasePath: undefined,
   /**
-   * Create a request builder for execution of get requests to the 'test/{id}' endpoint.
+   * Create a request builder for execution of get requests to the '/test/{id}' endpoint.
    * @param id - Path parameter.
    * @param queryParameters - Object containing the following keys: queryParam.
    * @param headerParameters - Object containing the following keys: headerParam.
    * @returns The request builder, use the \`execute()\` method to trigger the request.
    */
   getFn: (id: string, queryParameters: {'queryParam': QueryParameterType}, headerParameters?: {'headerParam'?: string}) => new OpenApiRequestBuilder<string>(
     'get',
-    "test/{id}",
+    "/test/{id}",
     {
           pathParameters: { id },
           queryParameters,
           headerParameters
-        }
+        },
+    TestApi._defaultBasePath
   ),
   /**
-   * Create a request builder for execution of post requests to the 'test' endpoint.
+   * Create a request builder for execution of post requests to the '/test' endpoint.
    * @param body - Request body.
    * @returns The request builder, use the \`execute()\` method to trigger the request.
    */
   createFn: (body: RefType) => new OpenApiRequestBuilder<ResponseType>(
     'post',
-    "test",
+    "/test",
     {
           body
-        }
+        },
+    TestApi._defaultBasePath
+  )
+};"
+`;
+
+exports[`api-file serializes api file with multiple operations and references and base path 1`] = `
+"import { OpenApiRequestBuilder } from '@sap-cloud-sdk/openapi';
+import type { QueryParameterType, RefType, ResponseType } from './schema';
+/**
+ * Representation of the 'TestApi'.
+ * This API is part of the 'MyServiceName' service.
+ */
+export const TestApi = {
+  _defaultBasePath: '/base/path/to/service/',
+  /**
+   * Create a request builder for execution of get requests to the '/test/{id}' endpoint.
+   * @param id - Path parameter.
+   * @param queryParameters - Object containing the following keys: queryParam.
+   * @param headerParameters - Object containing the following keys: headerParam.
+   * @returns The request builder, use the \`execute()\` method to trigger the request.
+   */
+  getFn: (id: string, queryParameters: {'queryParam': QueryParameterType}, headerParameters?: {'headerParam'?: string}) => new OpenApiRequestBuilder<string>(
+    'get',
+    "/test/{id}",
```

Comment:
Replying to my [other comment](https://github.com/SAP/cloud-sdk-js/pull/5247/files#r1884047673) here, why no trailing slash in base path. Because `/base/path/to/service/` + `/test/{id}` is wrong. Although the test will pass because we don't care what user provides as a base path, it can be a random string. But to make the test really "correct" I would leave the trailing slash out.